### PR TITLE
fix: Create link previews for websites with invalid head (WEBAPP-6278)

### PR DESCRIPTION
**Pull request by ffflorian**
A website consisting only of 
```html
<head>
  <meta property="og:image" content="http://example.com/image.jpg">
```
is technically not valid HTML5 **but** it has a preview image. Browsers will also display it correctly.

### CODE REVIEW

File: electron/src/lib/openGraph.ts [hunk 1]
```diff
@@ -146,13 +146,15 @@ const fetchOpenGraphData = async (url: string): Promise<OpenGraphResult> => {
   };
 
   const body = await axiosWithContentLimit(axiosConfig, CONTENT_SIZE_LIMIT);
-  const [head] = body.match(/<head>[\s\S]*?<\/head>/) || [''];
+  const matches = body.match(/.*property="og:[^"]+".*/gim) || [''];
```
Comment (ffflorian) [inline electron/src/lib/openGraph.ts]: https://regex101.com/r/kwrYZD/1

File: electron/src/lib/openGraph.ts [hunk 2]
```diff
@@ -42,6 +42,8 @@ const bufferToBase64 = (buffer: Buffer, mimeType: string): string => {
 
 const fetchImageAsBase64 = async (url: string): Promise<string | undefined> => {
   const IMAGE_SIZE_LIMIT = 5e6; // 5MB
+  const parsedUrl = parseUrl(encodeURI(url));
+  const normalizedUrl = parsedUrl.protocol ? parsedUrl : parseUrl(`http://${url}`);
```
Comment (lipis) [inline electron/src/lib/openGraph.ts]: https? or this for the ones that don't have the protocol?

File: electron/src/lib/openGraph.ts (hunk 1 above)
Comment (AndyLnd) [reply electron/src/lib/openGraph.ts]: what about single quotes?
```suggestion
  const matches = body.match(/.*property=(["'])og:.+?\1.*/gim) || [''];
```

File: electron/src/lib/openGraph.ts (hunk 2 above)
Comment (ffflorian) [reply electron/src/lib/openGraph.ts]: This is for the ones that don't have a protocol.

**Pull request merged**